dist/ion-autocomplete.js

```diff
@@ -151,12 +152,26 @@ angular.module('ion-autocomplete', []).directive('ionAutocomplete', [
                     compiledTemplate.scope.removeItem = function (index) {
                         // remove the item from the selected items and create a copy of the array to update the model.
                         // See https://github.com/angular-ui/ui-select/issues/191#issuecomment-55471732
-                        compiledTemplate.scope.selectedItems.splice(index, 1);
+                        var removed = compiledTemplate.scope.selectedItems.splice(index, 1);
+                        if(removed) {
```

Comment:
I think you are able to remove the `if` clause as this method is only called if there are items in the list to be deleted with the specified `index`, so you could just write:

```javascript
var removedItem = compiledTemplate.scope.selectedItems.splice(index, 1)[0]
```